### PR TITLE
reinstated and fixed multi dc gce snitch test

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by rayakurl** [comment]
@ShlomiBalalis -  unittests failed, please check why

File: sdcm/sct_config.py [hunk 1]
```diff
@@ -1381,7 +1381,7 @@ class SCTConfiguration(dict):
     }
 
     multi_region_params = [
-        'region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader'
+        'region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader', 'gce_datacenter'
```
Comment (fruch) [inline sdcm/sct_config.py:1384]: why are you adding this to this list ?, i.e. this shouldn't be here, 
those params are those who can have different value for different region, clearly `gce_datacenter` isn't that.

File: test-cases/features/google-cloud-snitch-multi-dc.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# TODO: need to qualify
```
Comment (fruch) [inline test-cases/features/google-cloud-snitch-multi-dc.yaml:1]: if you have this test proven working, please remove this comment

File: vars/runCleanupResource.groovy [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 def call(Map params, String region){
     def current_region = initAwsRegionParam(params.region, region)
+    def current_gce_datacenter = ""
+    if (params.backend == "gce") {
```
Comment (fruch) [inline vars/runCleanupResource.groovy:6]: why this is needed, i.e. what's wrong with passing down params.gce_datacenter ? for supporting random ? 

regardless if `initAwsRegionParam` isn't something specific to AWS it should be renamed.

File: sdcm/sct_config.py (hunk 1 above)
Comment (fruch) [reply sdcm/sct_config.py:1384]: seems like the unittest are failing cause of this change:
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-4978/3/execution/node/36/log/

**Review by fruch — CHANGES_REQUESTED** [review]
* remove changes from `multi_region_params`
* clarify why the usage of `initAwsRegionParam` is needed for

File: sdcm/sct_config.py (hunk 1 above)
Comment (ShlomiBalalis) [reply sdcm/sct_config.py:1384]: Am I misunderstanding the meaning of the list? In a multi dc scenario gce_datacenter will have "a different value" for each region? Same as region_name, which is in the list.
If gce_datacenter isn't in the multi_region_param list, the job fails nearly instantly:
```
Traceback (most recent call last):
  File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 1446, in <module>
    cli()
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 966, in run_test
    unittest.main(module=None, argv=['python -m unittest', argv],
  File "/usr/local/lib/python3.10/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/usr/local/lib/python3.10/unittest/main.py", line 147, in parseArgs
    self.createTests()
  File "/usr/local/lib/python3.10/unittest/main.py", line 158, in createTests
    self.test = self.testLoader.loadTestsFromNames(self.testNames,
  File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/local/lib/python3.10/unittest/loader.py", line 198, in loadTestsFromName
    inst = parent(name)
  File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 309, in __init__
    self.test_config.set_multi_region(len(self.params.get('gce_datacenter').split()) > 1)
AttributeError: 'list' object has no attribute 'split'
```

File: snitch_test.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            result = list(session.execute('select * from system.peers'))
+
+        self.log.debug(result)
+        assert result, 'ERROR: system.peers should be not empty.'
+        self.log.info("PASS: system.peers isn't empty as expected")
+
+        backend = self.params.get("cluster_backend")
+        if backend == "gce":
+            self.check_nodetool_status_output_gce()
+        elif backend == "aws":
```
Comment (fruch) [inline snitch_test.py:42]: this test is named `test_google_cloud_snitch`, how is AWS related here ?

File: sdcm/sct_config.py (hunk 1 above)
Comment (fruch) [reply sdcm/sct_config.py:1384]: `multi_region_params` was designed for AWS usage, and it used also for verification that all those field has the same length (a check which is done only for AWS, and adding this breaks it)

also a side effect of something being in this list, is that it turn back into a space separated string (to support all the code paths the assume those to be a string)

So:

1) there should be a separated list for each backend, for the verification sake
2) `gce_datacenter` should be turn back into a string, after validation.

Comment (fruch) [reply sdcm/sct_config.py:1384]: something like this:
```diff
diff --git a/sdcm/sct_config.py b/sdcm/sct_config.py
index f4550070..4e41ea47 100644
--- a/sdcm/sct_config.py
+++ b/sdcm/sct_config.py
@@ -621,7 +621,7 @@ class SCTConfiguration(dict):
 
         # GCE config options
 
-        dict(name="gce_datacenter", env="SCT_GCE_DATACENTER", type=str,
+        dict(name="gce_datacenter", env="SCT_GCE_DATACENTER", type=str_or_list,
              help="Supported: us-east1 - means that the zone will be selected automatically or "
                   "you can mention the zone explicitly, for example: us-east1-b"),
 
@@ -1389,10 +1389,10 @@ class SCTConfiguration(dict):
         "k8s-eks": [sct_abs_path('defaults/aws_config.yaml'), sct_abs_path('defaults/k8s_eks_config.yaml')],
     }
 
-    multi_region_params = [
-        'region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader'
-    ]
-
+    _multi_region_params = {
+        "aws": ['region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader'],
+        "gce": ['gce_datacenter', 'n_db_nodes'],
+    }
     stress_cmd_params = [
         # this list is used for variouse checks against stress commands, such as:
         # 1. Check if all c-s profile files existing that are referred in the commands
@@ -1423,6 +1423,7 @@ class SCTConfiguration(dict):
         if backend:
             backend_config_files += self.defaults_config_files[str(backend)]
 
+        self.multi_region_params = self._multi_region_params.get(str(backend), [])
         # 1) load the default backend config files
         files = anyconfig.load(list(backend_config_files))
         anyconfig.merge(self, files)
@@ -1755,8 +1756,8 @@ class SCTConfiguration(dict):
         db_type = self.get('db_type')
         self._check_version_supplied(backend)
         self._check_per_backend_required_values(backend)
-        if backend in ['aws'] and db_type != 'cloud_scylla':
-            self._check_aws_multi_region_params()
+        if backend in ['aws', 'gce'] and db_type != 'cloud_scylla':
+            self._check_multi_region_params()
 
         self._verify_data_volume_configuration(backend)
 
@@ -1798,8 +1799,9 @@ class SCTConfiguration(dict):
             if opt['name'] in self:
                 self._validate_value(opt)
 
-    def _check_aws_multi_region_params(self):
+    def _check_multi_region_params(self):
         region_count = {}
+        region_key = {"aws": 'region_name', 'gce': 'gce_datacenter'}
         for opt in self.multi_region_params:
             val = self.get(opt)
             if isinstance(val, str):
@@ -1808,7 +1810,7 @@ class SCTConfiguration(dict):
                 region_count[opt] = len(val)
             else:
                 region_count[opt] = 1
-        if not all(region_count['region_name'] == x for x in region_count.values()):
+        if not all(region_count[region_key.get(self.get('cluster_backend'))] == x for x in region_count.values()):
             raise ValueError("not all multi region values are equal: \n\t{}".format(region_count))
 
     def _validate_seeds_number(self):
diff --git a/unit_tests/test_config.py b/unit_tests/test_config.py
index 49c1ca8f..37a41165 100644
--- a/unit_tests/test_config.py
+++ b/unit_tests/test_config.py
@@ -576,6 +576,16 @@ class ConfigurationTests(unittest.TestCase):  # pylint: disable=too-many-public-
             conf = sct_config.SCTConfiguration()
             conf.verify_configuration()
 
+    def test_20_gce_mutli_dc(self):
+        os.environ['SCT_CLUSTER_BACKEND'] = 'gce'
+        os.environ['SCT_GCE_DATACENTER'] = 'a b c'
+        os.environ['SCT_N_DB_NODES'] = '2 2 2'
+        os.environ['SCT_GCE_IMAGE_DB'] = "image"
+        conf = sct_config.SCTConfiguration()
+        conf.verify_configuration()
+
+        self.assertEqual(conf.get('gce_datacenter'), 'a b c')
+
 
 if __name__ == "__main__":
     unittest.main()
```

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@ShlomiBalalis why do we need this backported ?

**Comment by fruch** [comment]
@ShlomiBalalis please fix the tests, seem like some of them are failing.

File: utils/lint_test_cases.sh [hunk 5]
```diff
@@ -4,7 +4,19 @@ OUT=0
...
+echo "2 region multi dc yamls"
+SCT_GCE_IMAGE_DB=image SCT_GCE_DATACENTER="us-east1 us-west1" SCT_SCYLLA_REPO='http://downloads.scylladb.com.s3.amazonaws.com/rpm/centos/scylla-2021.1.repo' ./sct.py lint-yamls -b gce -i 'multi-dc-rolling,3h-multidc,multidc-4h,topology-changes,snitch-multi-dc,manager-regression-multiDC' -e 'docker,azure,shutdown'
+OUT=$(($OUT + $?))
+
+echo "3 region multi dc yamls"
+SCT_GCE_IMAGE_DB=image SCT_GCE_DATACENTER="us-east1 us-west1 eu-north1" SCT_SCYLLA_REPO='http://downloads.scylladb.com.s3.amazonaws.com/rpm/centos/scylla-2021.1.repo' ./sct.py lint-yamls -b gce -i '24h-multidc,large-cluster,counters-multidc' -e 'docker,azure,shutdown'
+OUT=$(($OUT + $?))
+
+# Files that are named "multiple-dc" but are single dc
+SCT_GCE_IMAGE_DB=image SCT_GCE_DATACENTER=us-east1 SCT_SCYLLA_REPO='http://downloads.scylladb.com.s3.amazonaws.com/rpm/centos/scylla-2021.1.repo' ./sct.py lint-yamls -b gce -i 'shutdown'
```
Comment (fruch) [inline utils/lint_test_cases.sh:19]: this one file, please state while one it is, and doesn't seem to be used anywhere, sound like it's need to be remove.

I'm o.k. with not covering it at all.

**Comment by fruch** [comment]
@ShlomiBalalis also you have one conflict to solve

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@ShlomiBalalis sure about the backport ?

**Comment by ShlomiBalalis** [comment]
> @ShlomiBalalis sure about the backport ?
> 
> from my POV this is a new test case, (wasn't run for a few years), is this a must for having in 2022.1 ?

Roy requested it

**Comment by fruch** [comment]
@roydahan pending you approval for backporting

**Comment by roydahan** [comment]
@ShlomiBalalis Please run it on master make sure it passes.
Then, approved for backport to 5.0 and 2022.1

**Pull request merged**